libsolidity/ast/Types.cpp

```diff
@@ -3423,12 +3423,10 @@ MemberList::MemberMap FunctionType::nativeMembers(ASTNode const* _scope) const
 	}
 	case Kind::DelegateCall:
 	{
-		auto const* functionDefinition = dynamic_cast<FunctionDefinition const*>(m_declaration);
-		solAssert(functionDefinition, "");
-		solAssert(functionDefinition->visibility() != Visibility::Private, "");
-		if (functionDefinition->visibility() != Visibility::Internal)
+		if (auto const* functionDefinition = dynamic_cast<FunctionDefinition const*>(m_declaration))
 		{
-			auto const* contract = dynamic_cast<ContractDefinition const*>(m_declaration->scope());
+			solAssert(functionDefinition->visibility() != Visibility::Private && functionDefinition->visibility() != Visibility::Internal, "");
```

Comment:
The `Visibility` values are well ordered and you can compare them like this:

```suggestion
			solAssert(functionDefinition->visibility() > Visibility::Internal);
```

We actually do that a lot.